DQM/TrackingMonitorClient/plugins/TrackingAnalyzer.cc

```diff
@@ -91,18 +89,16 @@ void TrackingAnalyser::beginJob() { nLumiSecs_ = 0; }
 // -- Begin Run
 //
 void TrackingAnalyser::beginRun(edm::Run const& run, edm::EventSetup const& eSetup) {
-  edm::LogInfo("TrackingAnalyser") << "TrackingAnalyser:: Begining of Run";
+  edm::LogInfo("TrackingAnalyser") << " Begining of Run";
 
-  if (verbose_)
-    std::cout << "[TrackingAnalyser::beginRun]" << std::endl;
   // Check latest Fed cabling and create TrackerMapCreator
-  unsigned long long cacheID = eSetup.get<SiStripFedCablingRcd>().cacheIdentifier();
-  if (m_cacheID_ != cacheID) {
-    m_cacheID_ = cacheID;
-    edm::LogInfo("TrackingAnalyser") << "TrackingAnalyser::beginRun: "
+  if (fedCablingWatcher_.check(eSetup)) {  //this should check if cabling record has changed
+    edm::LogInfo("TrackingAnalyser") << "beginRun: "
                                      << " Change in Cabling, recrated TrackerMap";
-    eSetup.get<SiStripFedCablingRcd>().get(fedCabling_);
-    eSetup.get<SiStripDetCablingRcd>().get(detCabling_);
+    edm::ESHandle<SiStripFedCabling> fedcabHandle = eSetup.getHandle(fedCablingToken_);
+    fedCabling_ = fedcabHandle.product();
+    edm::ESHandle<SiStripDetCabling> detcabHandle = eSetup.getHandle(detCablingToken_);
+    detCabling_ = detcabHandle.product();
```

Comment:
These could be simplified a bit
```suggestion
    fedCabling_ = &eSetup.getData(fedCablingToken_);
    detCabling_ = &eSetup.getData(detCablingToken_);
```

Reply:
Indeed. Thanks! will update.